### PR TITLE
Bug26812

### PR DESCRIPTION
**Pull request by asn-d6**

### CODE REVIEW

File: src/feature/rend/rendservice.c [hunk 1]
```diff
@@ -546,14 +546,48 @@ copy_service_on_prunning(rend_service_t *dst, rend_service_t *src)
   dst->n_intro_points_wanted = src->n_intro_points_wanted;
 }
 
+/** A SIGHUP occured and service pruning might have happened; some services
+ *  might not be available anymore. Check if any introduction circuits are now
+ *  useless and need to be cleaned up. */
+static void
+clean_intro_circs_after_service_prune(const smartlist_t *surviving_services) {
```
Comment (dgoulet-tor) [inline src/feature/rend/rendservice.c]: This is a fun `{` location? :P

Comment (dgoulet-tor) [inline src/feature/rend/rendservice.c]: Also, I think you can remove `service_` from the name, we are in `rendservice.c` so it is imply. I think just `clean_intro_circs_after_pruning()` will be more meaningful. No strong opinion.

File: src/feature/rend/rendservice.c [hunk 2]
```diff
@@ -546,11 +546,26 @@ copy_service_on_prunning(rend_service_t *dst, rend_service_t *src)
   dst->n_intro_points_wanted = src->n_intro_points_wanted;
 }
 
+/** Close the introduction circuits of <b>s</b>. This function does not remove
+ *  the intro points from the service. */
+static void
+close_intro_circs_of_service(const rend_service_t *s)
```
Comment (dgoulet-tor) [inline src/feature/rend/rendservice.c]: Again, we are in `rendservice.c`, the service is implied so I would go with `close_intro_circuits()`?

File: src/feature/rend/rendservice.c (hunk 1 above)
Comment (asn-d6) [reply src/feature/rend/rendservice.c]: Fixed in new branch.

Comment (asn-d6) [reply src/feature/rend/rendservice.c]: Fixed in new branch.

File: src/feature/rend/rendservice.c (hunk 2 above)
Comment (asn-d6) [reply src/feature/rend/rendservice.c]: Fixed in new branch.

**Pull request closed without merging**